src/commands/distribute/groups/download.ts

```diff
@@ -148,37 +148,38 @@ export default class DownloadBinaryFromDistributionGroupCommand extends AppComma
     }
   }
 
-  private getDirectoryPath(directoryPath: string): Promise<string> {
+  private async getDirectoryPath(directoryPath: string): Promise<string> {
     if (!_.isNil(directoryPath)) {
       const normalizedPath = Path.normalize(directoryPath);
 
       debug("Checking that specified directories exist and creating them if not");
-      return new Promise<string> ((resolve, reject) => {
-        MkDirP(normalizedPath, (error: NodeJS.ErrnoException) => {
-          if (!_.isNil(error)) {
-            if (error.code === "EEXIST") {
-              reject(failure(ErrorCodes.InvalidParameter, `file ${directoryPath} already exists - directory path is expected`));
-            } else {
-              debug(`Failed to create/access directory ${directoryPath} - ${inspect(error)}`);
-              reject(failure(ErrorCodes.Exception, `failed to create/access directory ${directoryPath}`));
-            }
-          } else {
-            resolve(normalizedPath);
-          }
-        });
-      });
+      try {
+        await mkdirp(normalizedPath);
+        return normalizedPath;
+      } catch (error) {
+        if (error.code === "EEXIST") {
+          throw failure(ErrorCodes.InvalidParameter, `file ${directoryPath} already exists - directory path is expected`);
+        } else {
+          debug(`Failed to create/access directory ${directoryPath} - ${inspect(error)}`);
+          throw failure(ErrorCodes.Exception, `failed to create/access directory ${directoryPath}`);
+        }
+      }
     } else {
       // using current working directory by default
       return Promise.resolve(cwd());
     }
   }
 
-  private getFileFullPath(passedFileName: string, directoryPath: string, appName: string, downloadUrl: string): string {
+  private getFilenameFromDownloadUrl(downloadUrl: string): string {
+    const filename = Url.parse(downloadUrl).pathname?.split("/").slice(-1)[0];
```

Comment:
Won't this select just a single character? The `slice(-1)` selects the last string from the array of strings, and then `[0]` selects the first character of that string.

Reply:
Ah no I'm stupid ^^ Slice returns another array of course. Can't you take `[-1]` from the array? Is that Ruby?

Reply:
You can't do [-1]. I asked StackOverflow for the one-liner.